components/capi/include/pluralrules.h

```diff
@@ -0,0 +1,46 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#pragma once
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "provider.h"
+#include "locale.h"
+
+// opaque
+typedef struct PluralRules PluralRules; 
+
+typedef struct {
+    PluralRules* rules;
+    bool success;
+} CreatePluralRulesResult;
+
+typedef enum {
+    PluralRuleType_Cardinal,
+    PluralRuleType_Ordinal
+} PluralRuleType;
+
+typedef enum {
+    PluralCategory_Zero,
+    PluralCategory_One,
+    PluralCategory_Two,
+    PluralCategory_Few,
+    PluralCategory_Many,
+    PluralCategory_Other,
+} PluralCategory;
+
+typedef struct {
+    uint64_t i;
+    size_t v;
+    size_t w;
+    uint64_t f;
+    uint64_t t;
+    size_t c;
+} PluralOperands;
+
+
+CreatePluralRulesResult plural_rules_create(Locale* locale, ErasedDataProvider* provider, PluralRuleType ty);
+PluralCategory plural_rules_select(const PluralRules* rules, PluralOperands op);
```

Comment:
You've written `PluralOperands` as a data bag (which I agree with, at least for now).  We definitely want it in the header file so it can be directly constructed from C, but in the function signature, do we want to pass by value, or should we pass by pointer, just like object types?

Reply:
I think pointer, to me this seems like idiomatic C:
```
  PluralOperands p;
  PluralCategory cat = plural_rules_select(rules, &p);
```

Reply:
I feel like it doesn't really matter all that much, and I typically try to pass Copy types by value unless they're huge (this one isn't huge)

Reply:
Maybe it's that I don't think of PluralOperands as a value type; I think of it as a bag of values, and I expect a bag of values to be passed by reference/pointer.

We probably don't need PluralOperands to implement Copy.

Reply:
I'm using it as a Copy type within the bindings though I could call `.clone()`. It's just that `.clone()` is typically considered to be a heavy operation

Reply:
I see, good point.  PluralOperands can implement Copy but still be passed by reference, right?

I guess it comes down to a personal preference, unless there are performance impacts (i.e., I would guess that passing a 64-bit pointer is faster than `Copy`ing a struct that is a bit larger than that)

Reply:
Yes, they are passed by reference